drivers/serial/serial_gdbstub.c

```diff
@@ -253,7 +253,8 @@ static ssize_t uart_gdbstub_receive(FAR void *priv, FAR void *buf,
  *
  ****************************************************************************/
 
-static ssize_t uart_gdbstub_send(FAR void *priv, FAR void *buf, size_t len)
+static ssize_t uart_gdbstub_send(FAR void *priv, FAR const void *buf,
```

Comment:
![image](https://github.com/user-attachments/assets/e7f245d6-c8b4-4dc3-a528-0b43026573b1)
compiler won't warning here?

Reply:
It will if `-pedantic-errors` is enabled or using MSVC etc. Changed to `const char*` once and for all.